src/AppBundle/Command/AddUserCommand.php

```diff
@@ -0,0 +1,265 @@
+            // see http://symfony.com/doc/current/components/console/console_arguments.html
+            ->addArgument('username', InputArgument::OPTIONAL, 'The username of the new user')
+            ->addArgument('password', InputArgument::OPTIONAL, 'The plain password of the new user')
+            ->addArgument('email', InputArgument::OPTIONAL, 'The email of the new user')
+            ->addOption('is-admin', null, InputOption::VALUE_NONE, 'If set, the user is created as an administrator')
+        ;
+    }
+
+    /**
+     * This method is executed before initialize() and execute(). Its purpose is
+     * to check if some of the options/arguments are missing and interactively
+     * ask the user for those values.
+     *
+     * This method is completely optional. If you are developing an internal console
+     * command, you probably should not implement this method because it requires
+     * quite a lot of work. However, if the command is meant to be used by external
+     * users, this method is a nice way to fall back and prevent errors.
+     */
+    protected function interact(InputInterface $input, OutputInterface $output)
+    {
+        if (null !== $input->getArgument('username') && null !== $input->getArgument('password') && null !== $input->getArgument('email')) {
+            return;
+        }
+
+        $output->writeln('');
+        $output->writeln('Add User Command Interactive Wizard');
+        $output->writeln('-----------------------------------');
+
+        $output->writeln(array(
+            '',
+            'If you prefer to not use this interactive wizard, provide the',
+            'arguments required by this command as follows:',
+            '',
+            ' $ php app/console app:add-user username password email@example.com',
+            '',
+        ));
+
+        $output->writeln(array(
+            '',
+            'Now we\'ll ask you for the value of all the missing command arguments.',
+            '',
+        ));
+
+        // See http://symfony.com/doc/current/components/console/helpers/questionhelper.html
+        $console = $this->getHelper('question');
+
+        // Ask for the username if it's not defined
+        if (null === $username = $input->getArgument('username')) {
+            $question = new Question(' > <info>Username</>: ');
+            $question->setValidator(function ($answer) {
+                if (empty($answer)) {
+                    throw new \RuntimeException('The username cannot be empty');
+                }
+
+                return $answer;
+            });
+            $question->setMaxAttempts(self::MAX_ATTEMPTS);
+
+            $username = $console->ask($input, $output, $question);
+            $input->setArgument('username', $username);
+        } else {
+            $output->writeln(' > <info>Username</>: '.$username);
+        }
+
+        // Ask for the password if it's not defined
+        if (null === $password = $input->getArgument('password')) {
```

Comment:
I know this is Symfony style, but don't you think it's more readable to do this:

```php
$password = ...;
if (null === $password) {
```